tests/test_gemmbench_mlir_gen.py

```diff
@@ -69,6 +73,130 @@ def test_n_t_f8_f32_f8():
     )
 
 
+def test_n_t_f16_f32_f16_dynamic_dim_M():
+    # From 'llama8b_prefill'
+    cfg = GemmConfig(
+        M=kDynamic,
+        N=14336,
+        K=4096,
+        tA="N",
+        tB="T",
+        operand_element_type="f16",
+        accumulator_element_type="f32",
+        result_element_type="f16",
+        runtime_dim=512,  # Unused, included for correctness
+    )
+    mlir = generate_mlir(cfg)
+    match_lines(
+        mlir,
+        [
+            "module {",
+            "func.func @main(%arg0: tensor<?x4096xf16>, %arg1: tensor<14336x4096xf16>) -> tensor<?x14336xf16> {",
+            "%cst = arith.constant 0.000000e+00 : f32",
+            "%c0 = arith.constant 0 : index",
+            "%dim = tensor.dim %arg0, %c0 : tensor<?x4096xf16>",
```

Comment:
To make this more realistic, we can also inject `util.assume` ops that give this some minimum size and divisibility. But this can land is a follow up PR.

Reply:
Yes, you can check out sdxl dispatches: https://github.com/nod-ai/iree-model-benchmark/tree/main/sdxl -- add `--iree-hal-dump-executable-files-to=dump`